src/main/groovy/com/github/jrubygradle/internal/GemVersionResolver.groovy

```diff
@@ -4,10 +4,52 @@ import org.gradle.api.artifacts.DependencyResolveDetails
 
 class GemVersionResolver {
 
-    private Map versions = [:]
-  
+    Map versions
+    private final Object configuration
+    private final Object logger
+
+    static void setup(Object project) {
+        project.configurations.each {
+            new GemVersionResolver(project.logger, it)
+        }
+    }
+
+    GemVersionResolver(Object logger, Object configuration) {
```

Comment:
Is there a compelling reason to have the logger and configuration objects so loosely typed?